php/Exchange.php

```diff
@@ -219,6 +219,81 @@ class Exchange {
         'zb',
     );
 
+    public static $camel_methods = array (
```

Comment:
I don't really think that listing them all out here is maintainable. Basically, my suggestion was addressing the need to list them in the first place. I think it has to be automated, and, if some of the file is generated, then we don't need the overhead of the lookup in runtime – because we can just generate very fast methods in compile-time.

Reply:
listing the methods is ok in the base class, but if we do it for derived classes we lose the one to one mapping of js to php. imo it adds alot of unneccessary code bloat to the lib (which makes it harder to understand). We can tell users that underscore methods are slightly preferred in php (even if you generate another method for each camelcase method it will run at the same speed as using __call). In both cases we enter another stack frame, but no computation is being done in `__call` or `__call_static` just constant time hashmap lookup operations.

Reply:
I don't disagree with generating methods for the _base class only._ Having `method_exists` will help users in a language where there is little other sanity.